src/Nest/Aggregations/AggregationContainer.cs

```diff
@@ -287,6 +290,29 @@ public void Accept(IAggregationVisitor visitor)
 		{
 			if (visitor.Scope == AggregationVisitorScope.Unknown) visitor.Scope = AggregationVisitorScope.Aggregation;
 			new AggregationWalker().Walk(this, visitor);
+		}
+
+		internal static ConcurrentDictionary<Type, Type> _aggregateNameCache = new ConcurrentDictionary<Type, Type>();
+		internal static void InjectMetadata(IAggregation aggregation)
+		{
+			aggregation.Meta = aggregation.Meta ?? new Dictionary<string, object>();

```

Comment:
instead of new()'ing a a dictionary here we could keep an default dict per aggregation type.